File: 09-11-2024_Minimum_Array_End.java
class Solution {
    public long minEnd(long n, long x) {
        long finalValue = x;
        long remaining = n - 1;
        long bitPos = 1;

        while (remaining != 0) {
            if ((x & bitPos) == 0) {
                finalValue |= (remaining & 1) * bitPos;
                remaining >>= 1;
            }
            bitPos <<= 1;
        }

        return finalValue;
    }
}

File: 09-11-2024_Minimum_Array_End.cpp
class Solution {
public:
    long long minEnd(int n, int x) {
        long long final_value = x;
        long long remaining = n - 1;
        long long bit_pos = 1;

        while (remaining) {
            if (!(x & bit_pos)) {
                final_value |= (remaining & 1) * bit_pos;
                remaining >>= 1;
            }
            bit_pos <<= 1;
        }

        return final_value;
    }
};
